src/lib/getPeerDependenciesFromRegistry.ts

```diff
@@ -22,14 +24,23 @@ async function getPeerDependenciesFromRegistry(packageMap: Index<VersionSpec>, o
     bar.render()
   }
 
-  return Object.entries(packageMap).reduce(async (accumPromise, [pkg, version]) => {
-    const dep = await packageManager.getPeerDependencies!(pkg, version)
-    if (bar) {
-      bar.tick()
-    }
-    const accum = await accumPromise
-    return { ...accum, [pkg]: dep }
-  }, {})
+  const peerDependencies: Index<Index<string>> = Object.entries(packageMap).reduce(
+    async (accumPromise, [pkg, version]) => {
+      const dep = await packageManager.getPeerDependencies!(pkg, nodeSemver.coerce(version)?.version ?? version)
+      if (bar) {
+        bar.tick()
+      }
+      const accum = await accumPromise
+      const newAcc: Index<Index<string>> = { ...accum, [pkg]: dep }
+      const circularData = isCircularPeer(newAcc, pkg)
+      if (circularData.isCircular) {
+        delete newAcc[pkg][circularData.offendingPackage]
```

Comment:
`isCircularPeer` looks good. My only concern is that it does not consider the peerDependency version itself. It assumes that if there is a cycle, then the package can be upgraded, regardless of what peerDependency is listed. Is that true?

Reply:
So honestly I don't know. All this does currently is remove a peer dependency if it's presence would have resulted in a circular dependency. It doesn't specifically force any updates, just allows the normal code to update if appropriate. In theory, this could allow an update which shouldn't have been allowed in a very specific situation where there's a circular peer dependency chain but the versions are configured in a certain way.

It's worth noting that it'll only remove one peer dependency, not all of them. So in the example of `vitest` and `@vitest/ui` which are peers of each other, it'll remove just one's peer of the other. So for example, `@vitest/ui` will still have the real peer of `vitest`, but `vitest`'s peer of `@vitest/ui` will be removed to break the cycle.

I think in most cases where there's circular peer dependencies it's because the versions of those packages need to be kept in-sync. So this should hopefully work. But it hurts my brain to think through this so I am struggling to think of all of the different use-cases where this could possibly not work.

Reply:
Thanks, that makes sense. Given the complexity, the difficulty in mapping out all cases, and the impossibility of knowing the real-world likelihood of these peerDependency combinations in the wild, I think it's best to move forward with the new logic as-is. I hate to introduce a possible regression, but it will be far easier to debug and reason about with a concrete use case in the future than speculating about it now.